### PR TITLE
MM-36172 - purchase modal empty address issue

### PR DESCRIPTION
**Pull request by pablovelezvidal**
#### Summary
There was a situation where the purchase modal was trying to show the payment information in the case where the customer has not it set already, which leaded with a component showing empty strings separated by commas. This PR makes sure that when the customer has no payment information, the fill in payment method form is shown.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-36172

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Review by stevemudie — APPROVED** [review]
Will test post merge

File: utils/cloud_utils.ts [hunk 1]
```diff
@@ -20,3 +21,17 @@ export function isCustomerCardExpired(customer?: CloudCustomer): boolean {
...
+    }
+    return Boolean(
+        billingDetails?.address &&
+        billingDetails?.city &&
+        billingDetails?.state &&
+        billingDetails?.country &&
+        billingDetails?.postalCode &&
+        billingDetails.name,
+    );
+}
```
Comment (marianunez) [inline utils/cloud_utils.ts:37]: I believe we already have a method for this here: https://github.com/mattermost/mattermost-webapp/blob/2f0cc730068c208d53b0e91c2c885db6b4e02a23/types/cloud/sku.ts#L23-L37

**Comment by marianunez** [comment]
@pablovelezvidal @stevemudie just a reminder that we do have the label to setup the test server for the PR connected to CWS for this type of changes where we can do an initial test before merging when applicable.

File: utils/cloud_utils.ts (hunk 1 above)
Comment (pablovelezvidal) [reply utils/cloud_utils.ts:37]: got it.

**Review by nickmisasi — APPROVED** [review]

**Pull request merged**